### PR TITLE
Limit timeout stream query to <= now, extra query to get next

### PR DESCRIPTION
**Pull request by DavidBoike**
@Particular/ravendb-persistence-maintainers

### CODE REVIEW

File: src/NServiceBus.RavenDB/Timeouts/TimeoutPersister.cs [hunk 1]
```diff
@@ -65,45 +65,35 @@ public TimeoutPersister()
...
-                        results.Add(new Tuple<string, DateTime>(enumerator.Current.Document.Id, dateTime));
-                    }
+                if (nextTimeout != null)
+                {
+                    nextTimeoutToExpire = nextTimeout.Time;
                 }
 
                 // Next execution is either now if we know we got stale results or at the start of the next chunk, otherwise we delay the next execution a bit
-                if (qhi.IsStale && results.Count == 0)
+                else if (qhi.IsStale && results.Count == 0)
```
Comment (weralabaj) [inline src/NServiceBus.RavenDB/Timeouts/TimeoutPersister.cs]: in the comment we say "if we know we got stale results **or** at the start of the next chunk" but the condition is **and**.

**Comment by weralabaj** [comment]
One little comment, otherwise LGTM. The tests pass locally.

**Comment by mauroservienti** [comment]
LGTM as well, once @DavidBoike is back this can be merged and release to production IMO

**Comment by DavidBoike** [comment]
Wires got crossed here and this fell through the cracks. 3.0.11 has already been released. Will rebase this to a new 3.0.12 branch.

**Pull request closed without merging**